src/nvim/msgpack_rpc/server.c

```diff
@@ -57,18 +58,37 @@ bool server_init(void)
 {
   ga_init(&servers, sizeof(Server *), 1);
 
-  bool must_free = false;
   const char *listen_address = os_getenv(LISTEN_ADDRESS_ENV_VAR);
   if (listen_address == NULL || *listen_address == NUL) {
-    must_free = true;
-    listen_address = (char *)vim_tempname();
+    os_setenv(LISTEN_ADDRESS_ENV_VAR, os_serveraddress(), 1);
   }
 
-  bool ok = (server_start(listen_address) == 0);
-  if (must_free) {
-    xfree((char *) listen_address);
+  return (server_start(os_getenv(LISTEN_ADDRESS_ENV_VAR)) == 0);
+}
+
+///
+/// Return default address for local server. The returned buffer
+/// is statically allocated.
```

Comment:
Maybe just `return default address for local server in a statically allocated buffer`.